### PR TITLE
drm: vc4: txp: Do not allow 24bpp formats when transposing

### PR DESCRIPTION
**Pull request by 6by9**
The hardware doesn't support transposing to 24bpp (RGB888/BGR888) formats. There's no way to advertise this through DRM, so block it from atomic_check instead.

https://forums.raspberrypi.com/viewtopic.php?t=380129

### CODE REVIEW

File: drivers/gpu/drm/vc4/vc4_txp.c [hunk 1]
```diff
@@ -272,6 +272,13 @@ static int vc4_txp_connector_atomic_check(struct drm_connector *conn,
 		return -EINVAL;
 	}
 
+	if (conn_state->rotation == (DRM_MODE_ROTATE_0 | DRM_MODE_TRANSPOSE) &&
```
Comment (dividuum) [inline drivers/gpu/drm/vc4/vc4_txp.c]: Not sure if I'm violating the API, but I've been using the transpose feature by just setting `DRM_MODE_TRANSPOSE` without explicitly or'ing `DRM_MODE_ROTATE_0` as not rotating seems to be the default when setting the rotation property to `0`. As such my old code still slipped by this condition. With the explicit  `DRM_MODE_ROTATE_0` also or'ed, the commit fails as expected.

File: drivers/gpu/drm/vc4/vc4_txp.c [hunk 2]
```diff
@@ -272,6 +272,13 @@ static int vc4_txp_connector_atomic_check(struct drm_connector *conn,
...
+	    (fb->format->format == DRM_FORMAT_RGB888 ||
+	     fb->format->format == DRM_FORMAT_BGR888)) {
+		DRM_DEBUG_KMS("24bpp formats not supported when tranposing\n");
```
Comment (dividuum) [inline drivers/gpu/drm/vc4/vc4_txp.c]: Typo "tranposing"

File: drivers/gpu/drm/vc4/vc4_txp.c (hunk 1 above)
Comment (6by9) [reply drivers/gpu/drm/vc4/vc4_txp.c]: Yes it's violating the API. Added an extra patch that should validate that.

File: drivers/gpu/drm/vc4/vc4_txp.c (hunk 2 above)
Comment (6by9) [reply drivers/gpu/drm/vc4/vc4_txp.c]: Fixed.

**Comment by dividuum** [comment]
Triggered both error cases and confirmed that using the API correctly works as expected. LGTM.

**Comment by 6by9** [comment]
Thank you for testing.

**Comment by dividuum** [comment]
Neat. Thanks everyone!

**Pull request merged**